### PR TITLE
Add phpstan configuration file

### PR DESCRIPTION
**Pull request by tmotyl**
This PR provides basic configuration file to run phpstan on openMage code base.
Right now it just whitelists few core modules to show that it's working.
allowing more modules requires validation and probably some fixes, so should be tackled in separate PR's.

TODO:
Help is needed in providing a github action which will run phpstan on every PR.

### Description (*)
Provides basic configuration for running phpstan

### Fixed Issues (if relevant)

Fixes: 
1. Fixes https://github.com/OpenMage/magento-lts/issues/365

### Manual testing scenarios (*)

1. pull
2. composer install
3. run  `vendor/bin/phpstan analyze -c .github/phpstan.neon`

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All automated tests passed successfully (all builds are green)
 - [ ] Add yourself to contributors list

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -35,5 +35,11 @@
       "email": "github-sr@hotmail.com",
       "role": "Maintainer"
     }
-  ]
+  ],
+  "require-dev": {
+    "macopedia/phpstan-magento1": "^1.0"
+  },
+  "extra": {
+    "magento-root-dir": "."
```
Comment (Flyingmana) [inline composer.json:43]: not sure if this is a good idea, or if it causes any unexpected problems

Comment (tmotyl) [reply composer.json:43]: No idea, this was added by composer plugin after ‚composer install’.
If its not needed I can remove it.

**Comment by Flyingmana** [comment]
I created a new PR as I want this changes directly in the lowest version branch.
In this step I also added it to our github actions with an optional run, which executes it with a higher level, and also for a non yet covered path (so we always have a current overview of what things need to improve for the next steps.)

**Comment by Flyingmana** [comment]
merged via #1875 1875

**Pull request closed without merging**